learning_assistant/api.py

```diff
@@ -217,6 +217,11 @@ def get_message_history(courserun_key, user, message_count):
 
     Returns a number of messages equal to the message_count value.
     """
-    message_history = LearningAssistantMessage.objects.filter(
-        course_id=courserun_key, user=user).order_by('-created')[:message_count]
+    # Explanation over the double reverse: This fetches the last message_count elements ordered by creating order DESC.
+    # The result is the last chat messages for that user and course but in inversed order, so in order to flip them
+    # its first turn into a list and then reversed.
+    # The alternative of getting the last message_count ordering the query in ascending created order is not possible
+    # since negative indexing is not supported.
```

Comment:
Sorry to nit pick, but I still don't think this explains why we are limiting the number of messages prior to re-ordering them. It would be helpful to explain that the indexing is equivalent to the sql `limit`, which is more efficient than retrieving all messages and then processing that query set.

Reply:
Just updated it. Let me know what you think.